spec/services/mosaic_service_spec.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+require "rails_helper"
+require "shrine/storage/s3"
+
+RSpec.describe MosaicService do
+  # calculate the path, including fingerprint
+  # check to see if it exists in our storage adapter
+  # if so, return the path
+  # if not, generate it and return the new path
+  describe "#path" do
+    context "when the file does not exist on the storage adapter" do
+      with_queue_adapter :inline
+      it "generates the file and returns the path" do
+        allow(MosaicGenerator).to receive(:new).and_call_original
+        raster_set = FactoryBot.create_for_repository(:raster_set_with_files, id: "331d70a5-4bd9-4a65-80e4-763c8f6b34fd")
+        generator = described_class.new(resource: raster_set)
+        path = generator.path
+        expect(File.exist?(path)).to be true
+        expect(MosaicGenerator).to have_received(:new)
+        # Cleanup mosaic file
+        File.delete(path)
```

Comment:
Suggestion: Delete the whole cloud directory in a before hook. If this test fails it'll make successive tests unreliable.